diff --git a/cx.js b/cx.js
index f743a8df9375d8919d473dacab9640c6e66752c1..3435cfb53589fff6f938364e0cc99dd9ff4b5e73 100644
--- a/cx.js
+++ b/cx.js
@@ -1,6 +1,7 @@
 'use strict'
 
 var HAS_OWN = {}.hasOwnProperty
+function kebabize(str) { return str.replaceAll(/(?<!^)([A-Z])/g, '-$1').toLowerCase() }
 
 function getClass(arg) {
   if (typeof arg === 'string') return arg
@@ -23,7 +24,7 @@ function getClass(arg) {
         /* istanbul ignore next */
         if (HAS_OWN.call(arg, s)) {
           if (className) className += ' '
-          className += s
+          className += kebabize(s)
         }
       }
     }
@@ -46,7 +47,7 @@ module.exports = function (arg0) {
           /* istanbul ignore next */
           if (HAS_OWN.call(arg0, str)) {
             if (className) className += ' '
-            className += str
+            className += kebabize(str)
           }
         }
       }
