diff --git a/lib/container.js b/lib/container.js
index 8b9e1ef8425805e186575c71d518ff4682788b65..db03aea08838651194a8b0622379b01f59fb3e1f 100644
--- a/lib/container.js
+++ b/lib/container.js
@@ -304,7 +304,6 @@ class Container extends Node {
       try {
         result = callback(child, i)
       } catch (e) {
-        throw child.addToError(e)
       }
       if (result !== false && child.walk) {
         result = child.walk(callback)
diff --git a/lib/node.js b/lib/node.js
index 88f382a1c90f59117bd75dfba156dfbb47909dc2..c6bc83584598dd5a2537dcfff90e61339f77c3c8 100644
--- a/lib/node.js
+++ b/lib/node.js
@@ -55,7 +55,6 @@ class Node {
   }
 
   addToError(error) {
-    error.postcssNode = this
     if (error.stack && this.source && /\n\s{4}at /.test(error.stack)) {
       let s = this.source
       error.stack = error.stack.replace(
