diff --git a/dist/parser.js b/dist/parser.js
index ada61582777805359a7da8daaa6d6f5b6b6d556a..1ec7bf969bf8d492cee36faffd745cd85a0323b7 100644
--- a/dist/parser.js
+++ b/dist/parser.js
@@ -128,7 +128,7 @@ var Parser = /*#__PURE__*/function () {
     var _this = this;
     return function (message, errorOptions) {
       if (typeof _this.rule === 'string') {
-        return new Error(message);
+        return;
       }
       return _this.rule.error(message, errorOptions);
     };
