diff --git a/esm/vanilla/utils.mjs b/esm/vanilla/utils.mjs
index ee732e47abf25d102e6b37fb80f0e0eb600fe0ee..8fee8de694c492789615ccaa4f81d21ccbcbafd6 100644
--- a/esm/vanilla/utils.mjs
+++ b/esm/vanilla/utils.mjs
@@ -274,6 +274,7 @@ function proxyMap(entries) {
     get [Symbol.toStringTag]() {
       return "Map";
     },
+    __proto__: Map.prototype,
     toJSON() {
       return new Map(this.entries());
     }
@@ -399,6 +400,7 @@ function proxySet(initialValues) {
         yield [value, value];
       }
     },
+    __proto__: Set.prototype,
     toJSON() {
       return new Set(this.values());
     },
diff --git a/vanilla/utils.js b/vanilla/utils.js
index 544b8018040cdc997bbef00b1b883eb3223b00bd..c6ab45fc60c8b41575eb848f11e2283c56068072 100644
--- a/vanilla/utils.js
+++ b/vanilla/utils.js
@@ -276,6 +276,7 @@ function proxyMap(entries) {
     get [Symbol.toStringTag]() {
       return "Map";
     },
+    __proto__: Map.prototype,
     toJSON() {
       return new Map(this.entries());
     }
@@ -401,6 +402,7 @@ function proxySet(initialValues) {
         yield [value, value];
       }
     },
+    __proto__: Set.prototype,
     toJSON() {
       return new Set(this.values());
     },
